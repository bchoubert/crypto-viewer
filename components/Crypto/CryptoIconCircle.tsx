import Colors from "@/assets/Colors";
import { FC, ReactNode, memo, useMemo } from "react";
import { StyleSheet, View } from "react-native";

interface CryptoIconCircleProps {
  children: ReactNode;
  color: string;
}

const CryptoIconCircle: FC<CryptoIconCircleProps> = memo(({
  children,
  color,
}) => {
  const styles = useMemo(() => StyleSheet.create({
    container: {
      backgroundColor: color,
      color: Colors.white,
      height: 36,
      width: 36,
      padding: 5,
      borderRadius: 18,
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center'
    }
  }), [color]);

  return (
    <View style={styles.container}>
      {children}
    </View>
  );
});

export default CryptoIconCircle;
