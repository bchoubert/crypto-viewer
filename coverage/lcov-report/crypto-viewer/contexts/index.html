
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for crypto-viewer/contexts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> crypto-viewer/contexts</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50.63% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>40/79</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/36</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">52.38% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>11/21</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">49.32% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>36/73</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="NavigationProvider.tsx"><a href="NavigationProvider.tsx.html">NavigationProvider.tsx</a></td>
	<td data-value="65.52" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 65%"></div><div class="cover-empty" style="width: 35%"></div></div>
	</td>
	<td data-value="65.52" class="pct medium">65.52%</td>
	<td data-value="29" class="abs medium">19/29</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="7" class="abs medium">5/7</td>
	<td data-value="62.96" class="pct medium">62.96%</td>
	<td data-value="27" class="abs medium">17/27</td>
	</tr>

<tr>
	<td class="file low" data-value="SettingsProvider.tsx"><a href="SettingsProvider.tsx.html">SettingsProvider.tsx</a></td>
	<td data-value="10.53" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 10%"></div><div class="cover-empty" style="width: 90%"></div></div>
	</td>
	<td data-value="10.53" class="pct low">10.53%</td>
	<td data-value="19" class="abs low">2/19</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="11.76" class="pct low">11.76%</td>
	<td data-value="17" class="abs low">2/17</td>
	</tr>

<tr>
	<td class="file low" data-value="ThemeProvider.tsx"><a href="ThemeProvider.tsx.html">ThemeProvider.tsx</a></td>
	<td data-value="47.83" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 47%"></div><div class="cover-empty" style="width: 53%"></div></div>
	</td>
	<td data-value="47.83" class="pct low">47.83%</td>
	<td data-value="23" class="abs low">11/23</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="23" class="abs low">0/23</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="45.45" class="pct low">45.45%</td>
	<td data-value="22" class="abs low">10/22</td>
	</tr>

<tr>
	<td class="file high" data-value="TranslationProvider.tsx"><a href="TranslationProvider.tsx.html">TranslationProvider.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Mon Sep 06 2021 22:37:01 GMT+0200 (Central European Summer Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    