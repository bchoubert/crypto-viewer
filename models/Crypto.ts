// Currency type as returned by the API
export default interface Crypto {
  id: string,
  name: string,
  status: 'online' | 'offline',
  details: {
    type: string,
    symbol: string
  },
  price?: number
}
