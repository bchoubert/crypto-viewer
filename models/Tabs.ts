
export type tabType = 'list' | 'wallet' | 'settings' | 'details';

const Tabs = {
  list: 'list' as tabType,
  details: 'details' as tabType,
  settings: 'settings' as tabType,
  wallet: 'wallet' as tabType,
}

export default Tabs;
