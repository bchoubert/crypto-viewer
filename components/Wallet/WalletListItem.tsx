import { quoteDetails } from "@/types/crypto.types";
import { FC, memo, useCallback, useContext, useMemo } from "react";
import { Pressable, StyleSheet, Text, View } from "react-native";
import CryptoIcon from "../Utils/CryptoIcon";
import ECrypto from "@/constants/cryptos.enum";
import Colors, { EColor } from "@/assets/Colors";
import CryptoDetails from "@/constants/cryptodetails.constants";
import { printNumber } from '@/services/print.service';
import { Link } from "expo-router";
import Icon, { EIcon } from "../Utils/Icon";
import { SettingsContext } from "@/contexts/settings.provider";
import CryptoIconCircle from "../Utils/CryptoIconCircle";
import { WalletItem } from "@/types/wallet.types";
import { CryptoContext } from "@/contexts/crypto.provider";
import { TranslationsContext } from "@/contexts/translations.provider";

interface WalletListItemProps {
  item: WalletItem;
  setSelectedCrypto: (c: string) => void;
}

const styles = StyleSheet.create({
  container: {
    height: 60,
    borderBottomColor: Colors.lightGray,
    borderBottomWidth: 1,
    paddingLeft: 10,
    paddingRight: 10,
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    margin: 0,
    justifyContent: 'space-between',
    backgroundColor: Colors.white,
    position: 'relative',
  },

  swipeActions: {
    position: 'absolute',
    height: 60,
    left: '100%',
    top: 0,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
  },
  swipeActionsButton: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    width: 70,
    height: '100%',
  },
  swipeActionsButtonText: {
    color: Colors.white,
  },
  edit: {
    backgroundColor: Colors.blue,
  },
  delete: {
    backgroundColor: Colors.red,
  },

  iconplaceholder: {
    width: 36,
  },
  left: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    gap: 8,
  },
  right: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    gap: 5
  },
  numbers: {
    display: 'flex',
    flexDirection: 'column',
    width: 50,
  },
  quantity: {
    color: Colors.gray,
  },
  icon: {
    height: 20,
    width: 20,
    color: Colors.white,
  },
  titles: {
    display: 'flex',
    flexDirection: 'column',
  },
  title: {
    fontWeight: '500',
  },
  subtitle: {
    color: Colors.gray,
  }
})

const WalletListItem: FC<WalletListItemProps> = memo(({
  item,
  setSelectedCrypto,
}) => {
  const { settings, removeWalletItem } = useContext(SettingsContext);
  const { cryptos } = useContext(CryptoContext);
  const translation = useContext(TranslationsContext);

  const crypto = useMemo(() => cryptos.find(c => c.id === item.id)!, [item, cryptos]);

  const CryptoIconInstance = useMemo(() => CryptoIcon[item.id.toLowerCase() as ECrypto], [item.id]);
  const cryptoDetails = useMemo(() => CryptoDetails[item.id.toLowerCase() as ECrypto], [item.id]);

  const [color, quoteSymbol, statusColor] = useMemo(() => {
    const baseColor = cryptoDetails?.color || Colors.darkGray;
    const quoteSymbol = quoteDetails[settings.quote]?.symbol;
    const statusColor = crypto?.status === 'online' ? EColor.green : EColor.red;

    return [
      baseColor,
      quoteSymbol,
      statusColor,
    ]
  }, [cryptoDetails, settings, item]);

  const [quantity, amount] = useMemo(() => [
    item.quantity,
    crypto?.rate ? item.quantity * crypto.rate : 0,
  ], [crypto, item]);

  const deleteProxy = useCallback(() => {
    removeWalletItem(item.id);
  }, [removeWalletItem, item]);

  const setSelectedCryptoProxy = useCallback(() => {
    setSelectedCrypto(item.id);
  }, [item, setSelectedCrypto]);

  return (
    <>
      <Link href={`/crypto/${item.id}`} style={styles.container} asChild>
        <Pressable style={styles.container}>
          <View style={styles.left}>
            {CryptoIconInstance ? (
              <CryptoIconCircle color={color} statusColor={statusColor}>
                <CryptoIconInstance style={styles.icon} fill={Colors.white} />
              </CryptoIconCircle>
            ) : (<View style={styles.iconplaceholder} />)}
            <View style={styles.titles}>
              <Text style={styles.title}>{crypto.name}</Text>  
              <Text style={styles.subtitle}>{item.id}</Text> 
            </View>
          </View>
          <View style={styles.right}>
            <View style={styles.numbers}>
              <Text style={styles.quantity}>{printNumber(quantity, 'tokens')}</Text>
              {amount ? (
                <Text>{printNumber(amount, quoteSymbol)}</Text>
              ) : null}
            </View>
            <Icon name={EIcon.chevronRight} color={EColor.gray} width={12} />
          </View>
        </Pressable>
      </Link>
      <View style={styles.swipeActions}>
        <Pressable onPress={setSelectedCryptoProxy} style={{ ...styles.swipeActionsButton, ...styles.edit }}>
          <Text style={styles.swipeActionsButtonText}>{translation.common.edit}</Text>
        </Pressable>
        <Pressable onPress={deleteProxy} style={{ ...styles.swipeActionsButton, ...styles.delete }}>
          <Text style={styles.swipeActionsButtonText}>{translation.common.delete}</Text>
        </Pressable>
      </View>
    </>
  );
});

export default WalletListItem;