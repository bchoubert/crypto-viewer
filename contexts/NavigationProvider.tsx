import React, {
  createContext, FC, ReactNode, useCallback, useState, useEffect, useMemo, memo, useContext,
} from 'react';
import { StatusBarStyle, BackHandler } from 'react-native';
import { Colors } from 'react-native/Libraries/NewAppScreen';

import Tabs, { TabType } from '../models/Tabs';
import Crypto from '../models/Crypto';
import ColorService from '../services/Color.service';
import { ThemeContext } from './ThemeProvider';

export const NavigationContext = createContext({
  activeTab: Tabs.list as TabType,
  details: null as Crypto | null,
  statusBarColor: Colors.white as string,
  statusBarMode: 'dark-content' as StatusBarStyle,
  changeTab: null as (tabName: TabType, newDetails?: Object) => void | null,
});

interface NavigationProviderProps {
  children: ReactNode;
}

const NavigationProvider: FC<NavigationProviderProps> = ({ children }) => {
  const theme = useContext(ThemeContext);

  const [activeTab, setActiveTab] = useState<TabType>(Tabs.list);

  // Store details about current active tab
  const [details, setDetails] = useState<Crypto | null>(null);

  // StatusBar mode
  const [statusBarColor, setStatusBarColor] = useState<string>(theme.backgroundColor);
  const [statusBarMode, setStatusBarMode] = useState<StatusBarStyle>('dark-content');

  const changeTab = useCallback((tabName: TabType, newDetails: Crypto = null) => {
    setStatusBarColor(tabName === Tabs.details
      ? ColorService.getColorFromCrypto(newDetails.id)
      : theme.backgroundColor);
    setStatusBarMode(tabName === Tabs.details ? 'light-content' as StatusBarStyle : 'dark-content' as StatusBarStyle);
    setDetails(newDetails);
    setActiveTab(tabName);
  }, []);

  const handleBackButton = useCallback(() => {
    if ([Tabs.settings, Tabs.details].includes(activeTab)) {
      changeTab(Tabs.list);
      return true;
    }
    return false;
  }, [activeTab, changeTab]);

  useEffect(() => {
    const backHandler = BackHandler.addEventListener('hardwareBackPress', handleBackButton);
    return () => backHandler.remove();
  }, [activeTab]);

  const contextValues = useMemo(() => ({
    activeTab,
    details,
    statusBarColor,
    statusBarMode,
    changeTab,
  }), [activeTab, details, statusBarColor, statusBarMode, changeTab]);

  useEffect(() => {
    setStatusBarColor(theme.backgroundColor);
  }, [theme]);

  return (
    <NavigationContext.Provider value={contextValues}>
      {children}
    </NavigationContext.Provider>
  );
};

export default memo(NavigationProvider);
