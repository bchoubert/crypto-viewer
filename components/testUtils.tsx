/* eslint-disable import/no-extraneous-dependencies */
import { render, RenderAPI } from '@testing-library/react-native';
import '@testing-library/jest-dom/extend-expect';
import React, { ReactElement } from 'react';

import NavigationProvider from '../contexts/NavigationProvider';
import { SettingsContext } from '../contexts/SettingsProvider';
import ThemeProvider from '../contexts/ThemeProvider';
import TranslationProvider from '../contexts/TranslationProvider';

import { defaultSettings } from '../models/SettingsType';

jest.useFakeTimers();

export const enrichProviders = (node: ReactElement): ReactElement => (
  <SettingsContext.Provider
    value={{
      settings: defaultSettings,
      changeSettings: jest.fn(),
    }}
  >
    <TranslationProvider>
      <ThemeProvider>
        <NavigationProvider>
          {node}
        </NavigationProvider>
      </ThemeProvider>
    </TranslationProvider>
  </SettingsContext.Provider>
);

// eslint-disable-next-line max-len
export const renderNode = (node: ReactElement): RenderAPI => render(enrichProviders(node));
