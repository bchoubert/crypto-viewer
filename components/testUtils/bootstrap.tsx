/* eslint-disable no-console */
/* eslint-disable import/no-extraneous-dependencies */
import { render, RenderAPI } from '@testing-library/react-native';
import '@testing-library/jest-dom/extend-expect';
import React, { ReactElement } from 'react';
import * as PassThoughComponents from '../PassThroughComponents';

import NavigationProvider from '../../contexts/NavigationProvider';
import { SettingsContext } from '../../contexts/SettingsProvider';
import ThemeProvider from '../../contexts/ThemeProvider';
import TranslationProvider from '../../contexts/TranslationProvider';

import { defaultSettings } from '../../models/SettingsType';
import consoleMocker from './consoleMocker';

jest.useFakeTimers();

export const enrichProviders = (node: ReactElement): ReactElement => (
  <SettingsContext.Provider
    value={{
      settings: defaultSettings,
      changeSettings: jest.fn(),
    }}
  >
    <TranslationProvider>
      <ThemeProvider>
        <NavigationProvider>
          {node}
        </NavigationProvider>
      </ThemeProvider>
    </TranslationProvider>
  </SettingsContext.Provider>
);

// eslint-disable-next-line max-len
export const renderNode = (node: ReactElement): RenderAPI => render(enrichProviders(node));

beforeAll(() => {
  consoleMocker.mockConsole();
  // eslint-disable-next-line no-extend-native
  Date.prototype.getTimezoneOffset = jest.fn(() => 0);
});

afterAll(() => {
  consoleMocker.restoreConsole();
  jest.resetAllMocks();
});

jest.spyOn(PassThoughComponents, 'Image');
(PassThoughComponents.Image as unknown as jest.Mock).mockImplementation(() => (<div>IMAGE</div>));
