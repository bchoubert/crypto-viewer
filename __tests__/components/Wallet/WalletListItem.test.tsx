import { fireEvent } from '@testing-library/react-native';
import React from 'react';

import { renderNode } from '../../../components/testUtils/bootstrap';
import WalletListItem from '../../../components/Wallet/WalletListItem';
import { NavigationContext } from '../../../contexts/NavigationProvider';

import Crypto from '../../../models/Crypto';
import Tabs from '../../../models/Tabs';

describe('<WalletListItem />', () => {
  const editFromWalletFn = jest.fn();
  const deleteFromWalletFn = jest.fn();

  const changeTabFn = jest.fn();

  const commonProps = {
    cryptos: [
      {
        id: 'BTC',
        name: 'Bitcoin',
        price: 2,
        details: {
          symbol: 'B',
        },
      } as Crypto,
    ],
    walletItem: {
      crypto: 'BTC',
      amount: 3.5,
    },
    editFromWallet: editFromWalletFn,
    deleteFromWallet: deleteFromWalletFn,
  };

  beforeAll(() => {
    // eslint-disable-next-line no-console -- we don't need warn message from animate
    console.warn = () => null;
  });

  it('Prints correctly', () => {
    const { toJSON } = renderNode(
      <WalletListItem
        {...commonProps}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('Basic behaviour', () => {
    const node = renderNode(
      <NavigationContext.Provider
        value={{
          activeTab: 'wallet',
          changeTab: changeTabFn,
          details: null,
          statusBarColor: '',
          statusBarMode: 'default',
          handleBackAction: () => null,
        }}
      >
        <WalletListItem
          {...commonProps}
        />
      </NavigationContext.Provider>,
    );

    // Unit price
    node.getByText('3.50 B');
    // Item total
    node.getByText('7.00 $');

    node.getByTestId('Edit item Bitcoin');
    node.getByTestId('Delete item Bitcoin');

    expect(editFromWalletFn).toHaveBeenCalledTimes(0);
    expect(deleteFromWalletFn).toHaveBeenCalledTimes(0);

    fireEvent.press(node.getByTestId('Edit item Bitcoin'));
    expect(editFromWalletFn).toHaveBeenCalledTimes(1);
    expect(editFromWalletFn).toHaveBeenCalledWith('BTC');

    fireEvent.press(node.getByTestId('Delete item Bitcoin'));
    expect(deleteFromWalletFn).toHaveBeenCalledTimes(1);
    expect(deleteFromWalletFn).toHaveBeenCalledWith('BTC');

    // Open crypto
    expect(changeTabFn).toHaveBeenCalledTimes(0);
    fireEvent.press(node.getByText('Bitcoin'));
    expect(changeTabFn).toHaveBeenCalledTimes(1);
    expect(changeTabFn).toHaveBeenCalledWith(Tabs.details, commonProps.cryptos[0]);
  });

  it('Total', () => {
    const node = renderNode(
      <WalletListItem
        {...commonProps}
        walletItem={{
          crypto: 'total',
          amount: 56.23,
        }}
      />,
    );

    node.getByText('Total');
    node.getByText('56.23 $');
  });

  it('No crypto', () => {
    renderNode(
      <WalletListItem
        {...commonProps}
        walletItem={{
          crypto: 'NON_EXISTING_CRYPTO',
          amount: 2.5,
        }}
      />,
    );
  });

  it('Crpyto color', () => {
    const specificProps = {
      cryptos: [
        {
          id: 'btc',
          name: 'Bitcoin',
          price: 2,
          details: {
            symbol: 'B',
          },
        } as Crypto,
      ],
      walletItem: {
        crypto: 'btc',
        amount: 3.5,
      },
      editFromWallet: editFromWalletFn,
      deleteFromWallet: deleteFromWalletFn,
    };

    renderNode(
      <WalletListItem
        {...specificProps}
      />,
    );
  });
});
