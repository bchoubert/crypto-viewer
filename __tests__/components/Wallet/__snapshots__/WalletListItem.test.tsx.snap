// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WalletListItem /> Prints correctly 1`] = `
<View
  bounceOnMount={true}
  disable={false}
  leftActionActivationDistance={125}
  leftActionReleaseAnimationConfig={null}
  leftActionReleaseAnimationFn={null}
  leftButtonWidth={75}
  leftButtonsActivationDistance={75}
  leftButtonsCloseReleaseAnimationConfig={null}
  leftButtonsCloseReleaseAnimationFn={null}
  leftButtonsOpenReleaseAnimationConfig={null}
  leftButtonsOpenReleaseAnimationFn={null}
  onLayout={[Function]}
  onLeftActionActivate={[Function]}
  onLeftActionComplete={[Function]}
  onLeftActionDeactivate={[Function]}
  onLeftActionRelease={[Function]}
  onLeftButtonsActivate={[Function]}
  onLeftButtonsCloseComplete={[Function]}
  onLeftButtonsCloseRelease={[Function]}
  onLeftButtonsDeactivate={[Function]}
  onLeftButtonsOpenComplete={[Function]}
  onLeftButtonsOpenRelease={[Function]}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onPanAnimatedValueRef={[Function]}
  onRef={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onRightActionActivate={[Function]}
  onRightActionComplete={[Function]}
  onRightActionDeactivate={[Function]}
  onRightActionRelease={[Function]}
  onRightButtonsActivate={[Function]}
  onRightButtonsCloseComplete={[Function]}
  onRightButtonsCloseRelease={[Function]}
  onRightButtonsDeactivate={[Function]}
  onRightButtonsOpenComplete={[Function]}
  onRightButtonsOpenRelease={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onSwipeComplete={[Function]}
  onSwipeMove={[Function]}
  onSwipeRelease={[Function]}
  onSwipeStart={[Function]}
  rightActionActivationDistance={125}
  rightActionReleaseAnimationConfig={null}
  rightActionReleaseAnimationFn={null}
  rightButtonWidth={75}
  rightButtonsActivationDistance={75}
  rightButtonsCloseReleaseAnimationConfig={null}
  rightButtonsCloseReleaseAnimationFn={null}
  rightButtonsOpenReleaseAnimationConfig={null}
  rightButtonsOpenReleaseAnimationFn={null}
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
  swipeReleaseAnimationConfig={
    Object {
      "duration": 250,
      "easing": [Function],
      "toValue": Object {
        "x": 0,
        "y": 0,
      },
    }
  }
  swipeReleaseAnimationFn={[Function]}
  swipeStartMinDistance={15}
>
  <View
    style={
      Object {
        "flex": 1,
        "transform": Array [
          Object {
            "translateX": 0.5,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "height": 75,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7931B15",
            "borderRadius": 10,
            "flex": 1,
            "flexDirection": "row",
            "height": 62,
            "justifyContent": "space-between",
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#F7931B",
                "borderRadius": 10,
                "marginRight": 10,
              }
            }
          >
            <div>
              IMAGE
            </div>
          </View>
          <View
            style={
              Object {
                "fontSize": 25,
                "justifyContent": "flex-start",
                "width": 200,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#222222",
                  "fontSize": 18,
                }
              }
            >
              Bitcoin
            </Text>
            <Text
              style={
                Object {
                  "color": "#222222",
                  "fontSize": 12,
                }
              }
            >
              BTC
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#222222",
                  "paddingRight": 5,
                }
              }
            >
              3.50 B
            </Text>
            <Text
              style={
                Object {
                  "color": "#222222",
                  "paddingRight": 5,
                }
              }
            >
              7.00 $
            </Text>
          </View>
          <Text
            style={
              Object {
                "color": "#222222",
                "fontFamily": "crypto-viewer",
                "fontSize": 14,
              }
            }
          >
            î †
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginRight": -0,
        "transform": Array [
          Object {
            "translateX": 0.5,
          },
        ],
        "width": 0,
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "width": 0,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#005BE3",
            "borderBottomLeftRadius": 10,
            "borderTopLeftRadius": 10,
            "flex": 1,
            "justifyContent": "center",
          }
        }
        testID="Edit item Bitcoin"
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "paddingLeft": 26,
              "textAlign": "left",
            }
          }
        >
          Edit
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "width": 0,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#FF0D00",
            "borderBottomLeftRadius": 10,
            "borderTopLeftRadius": 10,
            "flex": 1,
            "justifyContent": "center",
          }
        }
        testID="Delete item Bitcoin"
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "paddingLeft": 20,
              "textAlign": "left",
            }
          }
        >
          Delete
        </Text>
      </View>
    </View>
  </View>
</View>
`;
